File: Leetcode/Leetcode-DSA-course/Hashing/2331-intersection-of-multiple-arrays/intersection-of-multiple-arrays.ts
function intersection(nums: number[][]): number[] {
    const counts = new Map();
    for (let arr of nums) {
        for (let x of arr)
            counts.set(x, (counts.get(x) || 0) + 1);
    }
    let n = nums.length;
    let ans = [];
    for (const [key, value] of counts) {
        if (value == n)
            ans.push(key);
    }
    ans.sort((a, b) => a - b);
    return ans;
};

File: Leetcode/Leetcode-DSA-course/Hashing/2331-intersection-of-multiple-arrays/intersection-of-multiple-arrays.java
class Solution {
    public List<Integer> intersection(int[][] nums) {
        Map<Integer, Integer> counts = new HashMap<>();
        for (int[] arr : nums) {
            for (int x : arr)
                counts.put(x, counts.getOrDefault(x, 0) + 1);
        }
        List<Integer> ans = new ArrayList<>();
        int n = nums.length;
        for (int key : counts.keySet()) {
            if (counts.get(key) == n) {
                ans.add(key);
            }
        }

        Collections.sort(ans);
        return ans;
    }
}